### PR TITLE
Test | Removing internal build of XunitExtensions and using the nuget

### PR DESCRIPTION
**Pull request by JRahnama**
This PR drops the internal build of Microsoft.DotNet.XUnitExtensions and uses the nuget from dotnet/arcade.

Also there is a version update on xunit from 2.4.2 to 2.5.0

Also Microsoft.Net.Test.SDK version is upgraded to 17.7.2

### CODE REVIEW

File: src/Microsoft.Data.SqlClient/tests/ManualTests/Microsoft.Data.SqlClient.ManualTesting.Tests.csproj [hunk 1]
```diff
@@ -307,6 +304,15 @@
...
     <PackageReference Condition="$(ReferenceType.Contains('Package'))" Include="Microsoft.Data.SqlClient" Version="$(TestMicrosoftDataSqlClientVersion)" />
+    <PackageReference Condition="$(TargetGroup) == 'netfx'" Include="System.Runtime.InteropServices.RuntimeInformation" Version="$(SystemRuntimeInteropServicesRuntimeInformationVersion)" />
+    <PackageReference Include="xunit" Version="$(XunitVersion)" />
+    <PackageReference Include="xunit.runner.visualstudio" Version="$(xunitrunnervisualstudioVersion)">
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+      <PrivateAssets>all</PrivateAssets>
+    </PackageReference>
+    <PackageReference Include="xunit.runner.console" Version="$(XunitVersion)" />
+    <PackageReference Include="xunit.runner.utility" Version="$(XunitVersion)" />
+    <PackageReference Include="Microsoft.DotNet.XUnitExtensions" Version="8.0.0-beta.23255.2" />
```
Comment (cheenamalhotra) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/Microsoft.Data.SqlClient.ManualTesting.Tests.csproj:315]: Use variable

File: src/Microsoft.Data.SqlClient/tests/FunctionalTests/Microsoft.Data.SqlClient.Tests.csproj [hunk 2]
```diff
@@ -64,9 +64,19 @@
...
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="$(MicrosoftNETTestSdkVersion)" />
+    <PackageReference Condition="$(TargetGroup) == 'netfx'" Include="System.Runtime.InteropServices.RuntimeInformation" Version="$(SystemRuntimeInteropServicesRuntimeInformationVersion)" />
+    <PackageReference Include="xunit" Version="$(XunitVersion)" />
+    <PackageReference Include="xunit.runner.visualstudio" Version="$(xunitrunnervisualstudioVersion)">
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+      <PrivateAssets>all</PrivateAssets>
+    </PackageReference>
+    <PackageReference Include="xunit.runner.console" Version="$(XunitVersion)" />
+    <PackageReference Include="xunit.runner.utility" Version="$(XunitVersion)" />
+    <PackageReference Include="Microsoft.DotNet.XUnitExtensions" Version="8.0.0-beta.23255.2" />
```
Comment (cheenamalhotra) [inline src/Microsoft.Data.SqlClient/tests/FunctionalTests/Microsoft.Data.SqlClient.Tests.csproj:79]: Use variable

**Comment by cheenamalhotra** [comment]
> This PR drops the internal build of Microsoft.DotNet.XUnitExtensions and uses the nuget from dotnet/arcade.

Can you be more specific how does pipeline access this package? I don't see you added any feed reference. Are all arcade packages available in upstream feed (dnceng/public)?

**Comment by JRahnama** [comment]
> > This PR drops the internal build of Microsoft.DotNet.XUnitExtensions and uses the nuget from dotnet/arcade.
> 
> Can you be more specific how does pipeline access this package? I don't see you added any feed reference. Are all arcade packages available in upstream feed (dnceng/public)?

This was not ready for review yet and was meant to be a POC and due to the same reason code cleaning was not done properly, but finally I had time to go through it and make it ready for review.

The Microsoft.DotNet.XUnitExtensions should be available from dnceng according to its [documentation](https://github.com/dotnet/arcade#getting-started).

**Review by David-Engel — APPROVED** [review]

**Review by David-Engel — APPROVED** [review]

File: src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/LocalDBTest/LocalDBTest.cs [hunk 3]
```diff
@@ -26,15 +26,15 @@ private enum InfoType
         private static readonly string s_startLocalDbCommand = @$"/c SqlLocalDb start {DataTestUtility.LocalDbAppName}";
         private static readonly string s_localDbNamedPipeConnectionString = @$"server={GetLocalDbNamedPipe()}";
 
+#if !WINDOWS_UAP
         #region LocalDbTests
-        [SkipOnTargetFramework(TargetFrameworkMonikers.Uap)] // No Registry support on UAP
+
```
Comment (DavoudEshtehari) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/LocalDBTest/LocalDBTest.cs:31]: Replacing this attribute with a directive will skip showing these tests in the report in unsupported TF. There should be a solution to detect it in a utility. [PalatformDetection](https://github.com/dotnet/runtime/blob/f7a7b2b8b12dac91aa86a31e59ef6c1028d7113e/src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs) from runtime would be helpful.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/TransactionTest/DistributedTransactionTest.cs [hunk 4]
```diff
@@ -14,7 +15,7 @@ namespace Microsoft.Data.SqlClient.ManualTesting.Tests
     [PlatformSpecific(TestPlatforms.Windows)]
     public class DistributedTransactionTest
     {
-        private static bool s_DelegatedTransactionCondition => DataTestUtility.AreConnStringsSetup() && DataTestUtility.IsNotAzureServer() && PlatformDetection.IsNotX86Process;
+        private static bool s_DelegatedTransactionCondition => DataTestUtility.AreConnStringsSetup() && DataTestUtility.IsNotAzureServer() && (RuntimeInformation.ProcessArchitecture != Architecture.X86);
```
Comment (DavoudEshtehari) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/TransactionTest/DistributedTransactionTest.cs:18]: NIT: I'd move it into DataTestUtility.

**Review by DavoudEshtehari — APPROVED** [review]

File: src/Microsoft.Data.SqlClient/tests/ManualTests/xunit.runner.json [hunk 5]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (DavoudEshtehari) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/xunit.runner.json:1]: Can we move this file to a shared location for both projects?

File: src/Microsoft.Data.SqlClient/tests/FunctionalTests/DataCommon/TestUtility.cs [hunk 6]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
```
Comment (DavoudEshtehari) [inline src/Microsoft.Data.SqlClient/tests/FunctionalTests/DataCommon/TestUtility.cs:1]:
```suggestion
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.
// See the LICENSE file in the project root for more information.

﻿using System;
```

File: src/Microsoft.Data.SqlClient/tests/ManualTests/DataCommon/DataTestUtility.cs [hunk 7]
```diff
@@ -1068,5 +1071,29 @@ public static string GetMachineFQDN(string hostname)
             }
             return fqdn.ToString();
         }
+
+        public static bool IsRunningAsUWPApp()
```
Comment (DavoudEshtehari) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/DataCommon/DataTestUtility.cs:1075]: You can cache the resolved value with the first call and use it with consecutive calls.

**Review by DavoudEshtehari — APPROVED** [review]

**Pull request merged**